browser/components/CodeEditor.js

```diff
@@ -644,6 +626,55 @@ export default class CodeEditor extends React.Component {
     }
   }
 
+  getCodeEditorLintConfig () {
+    const { mode } = this.props
+    const checkMarkdownNoteIsOpen = mode === 'Boost Flavored Markdown'
+
+    return checkMarkdownNoteIsOpen ? {
+      'getAnnotations': this.validatorOfMarkdown,
+      'async': true
+    } : false
+  }
+
+  validatorOfMarkdown (text, updateLinting) {
+    const { customMarkdownLintConfig } = this.props
+    let lintConfigJson
+    try {
+      Jsonlint.parse(customMarkdownLintConfig)
+      lintConfigJson = JSON.parse(customMarkdownLintConfig)
+    } catch (err) {
+      throw err
```

Comment:
I don't think throwing an error here is a good thing to do because it will make the app unusable. I think you should put something more "user-friendly" for example not applying the lint config and display error message above the custom lint config box in the setting.

Reply:
Yes, you're right. It's bad to throw an error with-out handling it.
How can I add the error message display? I've tried to change the `throw err` to 
```js
eventEmitter.emit('APP_SETTING_ERROR')
return
```

But I don't like how it's working. It first shows success message and then the error occured message. It happens pretty fast - short flicker of the green message. Also on a second click it won't show the error again.

The `catch` can be triggered by entering invalid json for the custom markdown lint rule and then press the save button.

Reply:
A custom error message wasn't working - not sure why. I wanted to add something like: `Failed to apply custom linting rule.`

Tried it with `eventEmitter.emit('APP_SETTING_ERROR', {message: 'Failed to apply...'})`